Silver Slime Core and Random Reagent changes

FalseIncarnate opened the pull request:
Silver Slime Cores now should only spawn drink containers that come pre-filled with a drink.
- This means no more empty cups/trophies/bottler containers/flasks

![image](https://user-images.githubusercontent.com/8988254/30463360-4220842a-9999-11e7-8317-b0ae64f8bc47.png)

The random reagent bottle now properly respects blacklisted chems, instead of blindly picking from all reagents.
- If the reagent can't be synthesized (can_synth = FALSE), it won't spawn in the bottle. This is now consistent with other sources of random chemicals, such as botany strange seeds.

Removed a file's worth of relative pathing (random_items.dm)
- Additionally adjusted the code for picking random reagents to use copied lists or `get_random_reagent_id()` as appropriate to keep things sanitized and respect blacklisted chems

:cl:
tweak: Silver Slime Core reactions can only summon forth drinks that actually contain a drink. Begone thirst!
tweak: Removes some "unsafe" chemicals from lists of potential chemicals for inclusion in things like random pills or bottles, added one to replace them.
tweak: Removes adminordrazine and nanites from vent clog, replaces nanites with syndicate_nanites.
/:cl:

uraniummeltdown commented:
Why remove xenomicrobes? Xenobiology is the only way to make them. And you could just make a rule that injecting yourself with them doesn't automatically make you an antag (but being a victim of someone else does). It's not like being xeno'd is irreversible, you can transfer a xeno brain to a human body

Birdtalon commented:
@uraniummeltdown He hasn't. Check line 58.

FalseIncarnate commented:
Self-antagging is generally something we want to avoid making possible. Yes, it makes you a hunter which cannot lay eggs and start an infestation, however it does still make you a full antag xeno.

Additionally, it takes just under 2 units of xenomicrobes to cause the transformation disease per person and the random bottles would have at least enough for 5 people to be infected. We made it so you can't turn yourself into a xeno through a facehugger to avoid people deliberately throwing themselves at the xenos to play one themselves, why would we keep a second way of making yourself a xeno?

uraniummeltdown commented:
Just make the xenomicrobe transformation give a notice to the user that if they weren't an antag before they aren't one now, or write that down in the rules section on self-antagging. Also you can raise the requirement for the xeno transformation

FalseIncarnate commented:
Because its not meant to be a commonly obtainable reagent. All I did was remove it from the lists of random chems used in generating random reagent bottles, random pills, and similar items. The reagent still exists, you just won't find it in a random bottle laying around in maint in 10+ unit quantities.

As for simply telling the victim of a xenomicrobe-triggered transformation that they aren't an antag, that's a bandaid over the issue. Xenos are not meant to be an everyday thing you can turn yourself in to because you feel like it. Antag or no, the issue is that you can turn yourself into a combat-oriented death machine that can't be disarmed of their weapons... aka a powergaming validhunter's ideal state. I'm not touching the effects of the xenomicrobes or their resulting transformation disease. I'm just removing one source of the reagent that shouldn't have been a source to begin with.

Bram961 commented:
"Commonly obtainable" no it isn't

"you just won't find it in a random bottle laying around in maint in 10+ unit quantities." Could have just said anywhere instead

"I'm just removing one source" the only source.

FalseIncarnate commented:
"Commonly available" was referring to literally anyone could obtain it, not just a specific job/role.

As for removing one source, I'm also going to be adding one source (originally was going to be in a separate PR to avoid scope-creep) which will make it still obtainable yet not as simple as "pick up bottle, make xenos" like how it was. Random reagents bottles/pills could be obtained outside of xenobiology's silver slime core reaction, so the assertion that xenobiology was the only source is in itself a falsehood.

I already have realized my error in thinking it was obtainable elsewhere, since I misread some code. While I admit and apologize for that mistake on my part, I do want to remind you that xenomicrobes are still something that should not be in the random pills or bottles like it was because of the power it has, which is the same reasoning those bottles likewise can't have things like venom or spider eggs.

I additionally realized today that I forgot to remove another source of admin chems, so I'll need to fix that tonight and will include the reintroduction of xenomicrobes elsewhere at the same time.

Bram961 commented:
There is no reason to remove it from xenobiology.

Jountax commented:
@uraniummeltdown uranium Technically, being transformed into a borg can also be cured, if that's a reasoning for keeping it in random pills and stuff. It's _slightly_ harder but not impossible.

But what does adminordrazine even do?

Bram961 commented:
Adminodrazine heals all damage types (at about 15 points for every unit used for every damage type) and cures any diseases/brokenbones/defects

mrturkeytoe commented:
All I can say about this PR and the bee PR is this is the exact same argument made against gatfruit and history has proven gatfruit are not abused and this is 100% hugbox, bubblewrapping fun removal.

imsxz commented:
You can only find the unlabled bottles in two places that I know of: xenobio and a certain gateway mission.
I don't think it's fair to say that "anyone regardless of role" can find xenomicrobes...

Jountax commented:
Oh, by all means the admin drug probably needs to be removed. Seems like the kind of thing that only admins should have access too, as the name suggests.

Landerlow commented:
The removal of adminorazine and other stuff is unnecessary. There's no need to keep removing stuff. Especially if it isn't abused.

FalseIncarnate commented:
Adminordrazine and nanites (literally just renamed adminordrazine) should not be available to players without admin intervention. The fact that it literally has "admin" in the name clearly states that it is for admins. It is a testing and debugging drug that resets all conditions, mutations, broken bones, clone damage, brain damage, and some other things, and as such shouldn't be something normally available since it isn't meant to be obtainable.

Xenomicrobes were removed from random reagent sources since they provided larger amounts than seemed reasonable for how low effort those sources were. I instead moved them to a low chance to appear in small amounts within xeno meat, so you have to put in the effort of finding, killing, and butchering a xeno in order to get a reagent that makes you into a full fledged xeno. Additionally, there is a small chance that a vent clog might spew out a cloud of xenomicrobes or nanomachines (among other things) in smaller dosages than normal vent clog reagents, to make the event more interesting than "well, I stood by a vent and now am ODing and addicted to meth"

Gatfruit is an unfair comparison and I'd like to invite everyone to stop attempting to make that comparison. Most of the concern with gatfruit stemmed from misunderstanding it's roundstart availability, not the ability to get a gun (which science has proven more than capable of without RNG).

This PR definitely bloated a bit from the initial intention, but I attempted to keep the changes related. I removed a chem from its only source and so I readied it elsewhere to keep it obtainable, and removed chems that shouldn't be obtainable from various sources. Additionally, I tweaked the silver core reaction to only provide filled drinks because getting 5 empty cups is a bit of a disappointment, and something that supplies drinks probably shouldn't be supplying random meds instead.

This never was about something being abused, but rather preventing the potential and clearing up some inconsistencies.

TullyBurnalot commented:
The Heads and Maintainers would prefer if the PR only blacklisted adminordrazine by removing it from the rare_chems list; we're not looking for any of the other changes.

ktoma36 commented:
@TullyBurnalot Do you guys also want the removal of admin booze from the list of drinks silver slime extract can make? In all my time of playing I have only see it come up twice. It contains 5 units of adminizone and a number of drugs and what not.

@FalseIncarnate Xenobio is the only way to get some of those cool cups, and if you are breeding silver slimes getting a few empty drinks isnt really an issue

FalseIncarnate commented:
Brit cups are available in the medical break room.
Silly cups (paper cups) are available in medical and maintenance.
Flasks are either available with their associated job, in the bar, or should not be publically available (fluff items)
Shakers are available in the bar or via bartending crate from cargo
Drinking glasses and shot glasses are available from bar vendor (plus would be left behind by drinking or emptying the beverages summoned)
Bottler bottles/cans are available from bottler or should not be available (base type)
I assume the mushroom bowl is something lavaland related or otherwise obtainable elsewhere

Quite frankly, the only ones not normally available are the ones that shouldn't be normally available (fluff, base types) or the trophies which are meant more as event rewards than standard drinking items.

I see no reason to keep empty containers in the possibilities of the reaction since they are already available and simply dilute the chances of getting something you can actually drink.

Likewise, I will keep the removal of the adminbooze and madmin malt drinks for the original reasoning (they contain admin-only chems). Badmin brew remains as it contains player-accessible chems only.

I will not be removing everything except for the rare_chems list changes from this PR as much of this is directly related to players being able to obtain admin reagents. I will however revert the xenomicrobes bits since those appear to be the biggest thorn to most people. I'll put those in a separate PR.

ktoma36 commented:
You origionally stated you were going to remove adminizone from the random reagents list. Admin booze is stupidly rare and contains a trace amount of adminizone, as I said 5 units. Also are you planning on removing the drinks entirely or just from the spawn list?

FalseIncarnate commented:
Just the spawn list as I already had. The drink bottles themselves exist for admins, and I see no reason to remove them entirely, just removing them from player-accessible methods is sufficient there.

I am still going to remove it from the random reagents list as was requested, I was (attempting, albeit unclearly I do admit) stating that I wasn't going to roll back every single change except for that list change. I'm only going to roll back the most controversial one (xenomicrobes) since the rest are in fact related to the initial topics of the PR

mrturkeytoe commented:
I'd be fine with this PR if you kept the random reagent bottles.

ktoma36 commented:
Maybe make the random bottles spawn with random bar drinks? Though that could make a multitude of more issues...

FalseIncarnate commented:
There is a random drink bottle (it's that skinny blue one in the original picture) that can only be filled with a random drink reagent (the one in the picture ended up being ice by pure coincidence). That does still spawn from the silver slime core.

FalseIncarnate commented on code/modules/reagents/chemistry/recipes/slime_extracts.dm:
```diff
@@ -193,6 +193,22 @@
 	var/list/borks = subtypesof(/obj/item/weapon/reagent_containers/food/drinks)
+	var/list/blocked = list(/obj/item/weapon/reagent_containers/food/drinks/cans/adminbooze,
+							/obj/item/weapon/reagent_containers/food/drinks/cans/madminmalt,
+							/obj/item/weapon/reagent_containers/food/drinks/bottle/random_reagent,
```
Reminder to self: remove this line for merge

FalseIncarnate commented:
@tigercat2000 this is ready to merge.

Removal of the random reagent bottle from the silver slime core has been removed from this PR, will be posted in its own PR at a later date.

Additionally, added another blacklisted item to the food reaction that someone missed: the deepfried food holder. This is a semi-abstract item that shouldn't exist in it's "standard" state, as it is a weird thing for the deepfryer (even it's description says that if you see it the deep fryer code broke).
